v2/types/backoff.go

```diff
@@ -0,0 +1,14 @@
+package types
+
+import "time"
+
+// Backoff holds parameters applied to retries
+type Backoff struct {
```

Comment:
Rather than make a custom type, I would hide that detail from the user with a change to the WithBackoff method, I will comment there.